node/service/src/lib.rs

```diff
@@ -294,417 +323,412 @@ fn real_overseer<S: SpawnNamed>(
 	).map_err(|e| ServiceError::Other(format!("Failed to create an Overseer: {:?}", e)))
 }
 
-/// Builds a new service for a full client.
-#[macro_export]
-macro_rules! new_full {
-	(
-		$config:expr,
-		$collating_for:expr,
-		$authority_discovery_disabled:expr,
-		$grandpa_pause:expr,
-		$runtime:ty,
-		$dispatch:ty,
-	) => {{
-		use sc_client_api::ExecutorProvider;
-		use sp_core::traits::BareCryptoStorePtr;
-
-		let is_collator = $collating_for.is_some();
-		let role = $config.role.clone();
-		let is_authority = role.is_authority() && !is_collator;
-		let force_authoring = $config.force_authoring;
-		let disable_grandpa = $config.disable_grandpa;
-		let name = $config.network.node_name.clone();
-
-		let (builder, mut import_setup, inherent_data_providers, mut rpc_setup) =
-			new_full_start!($config, $runtime, $dispatch);
-
-		let ServiceComponents {
-			client, network, select_chain, keystore, transaction_pool, prometheus_registry,
-			task_manager, telemetry_on_connect_sinks, ..
-		} = builder
-			.with_finality_proof_provider(|client, backend| {
-				let provider = client as Arc<dyn grandpa::StorageAndProofProvider<_, _>>;
-				Ok(Arc::new(GrandpaFinalityProofProvider::new(backend, provider)) as _)
-			})?
-			.build_full()?;
-
-		let (block_import, link_half, babe_link) = import_setup.take()
-			.expect("Link Half and Block Import are present for Full Services or setup failed before. qed");
-
-		let shared_voter_state = rpc_setup.take()
-			.expect("The SharedVoterState is present for Full Services or setup failed before. qed");
-
-		let overseer_client = client.clone();
-		let spawner = task_manager.spawn_handle();
-		let leaves: Vec<_> = select_chain.clone().ok_or(ServiceError::SelectChainRequired)?
-			.leaves()
-			.unwrap_or_else(|_| vec![])
-			.into_iter()
-			.filter_map(|hash| {
-				let number = client.number(hash).ok()??;
-				let parent_hash = client.header(&BlockId::Hash(hash)).ok()??.parent_hash;
-
-				Some(BlockInfo {
-					hash,
-					parent_hash,
-					number,
-				})
+
+fn new_full<RuntimeApi, Executor, Extrinsic>(
+	config: Configuration,
+	collating_for: Option<(CollatorId, ParaId)>,
+	_max_block_data_size: Option<u64>,
+	_authority_discovery_disabled: bool,
+	_fslot_duration: u64,
```

Comment:
I haven't been following all the updates, but what are these unused params for? can we just remove them?

Reply:
In the legacy service, `slot_duration` is used in `consensus::ProposerFactory::new`, so I think we can assume that they will be used at some point.